File: app/src/debug/java/com/handstandsam/shoppingapp/NetworkConfig.java
package com.handstandsam.shoppingapp;

import android.content.Context;

import com.handstandsam.shoppingapp.di.NetworkModule;
import com.handstandsam.shoppingapp.mockaccount.MockAccount;
import com.handstandsam.shoppingapp.mockaccount.Stubberator;
import com.handstandsam.shoppingapp.wiremock.WireMockManager;


class NetworkConfig {
    Context applicationContext;

    public NetworkConfig(Context applicationContext) {
        this.applicationContext = applicationContext;
    }

    public void startNormally() {
    }

    public void stubLocalWireMock(MockAccount mockAccount) {
        NetworkModule.USE_LOCAL_SERVER = true;
        new Stubberator(applicationContext).stubItAll(mockAccount);
    }

    public void stubRemoteWireMock(MockAccount mockAccount) {
        NetworkModule.USE_LOCAL_SERVER = false;
        new Stubberator(applicationContext).stubItAll(mockAccount);
    }


    public void recordMappingsAndProxy(String remoteBaseUrl) {
//                Record
        WireMockManager wireMockManager = new WireMockManager.Builder(applicationContext, NetworkModule.LOCALHOST_PORT).build();
        wireMockManager.startProxyAndRecord(remoteBaseUrl);
    }


    public void playbackRecordedMappings() {
//                Playback
        WireMockManager wireMockManager = new WireMockManager.Builder(applicationContext, NetworkModule.LOCALHOST_PORT).build();
        wireMockManager.startPlayBack();
    }
}

File: app/src/debug/java/com/handstandsam/shoppingapp/NetworkConfig.kt
package com.handstandsam.shoppingapp

import android.content.Context
import com.handstandsam.shoppingapp.di.NetworkConstants
import com.handstandsam.shoppingapp.mockaccount.Stubberator
import com.handstandsam.shoppingapp.mockdata.MockAccount
import com.handstandsam.shoppingapp.wiremock.WireMockManager


internal class NetworkConfig(var applicationContext: Context) {

    fun startNormally() {}

    fun stubLocalWireMock(mockAccount: MockAccount) {
        Stubberator(useLocalServer = true).stubItAll(mockAccount)
    }

    fun stubRemoteWireMock(mockAccount: MockAccount) {
        Stubberator(useLocalServer = false).stubItAll(mockAccount)
    }

    fun recordMappingsAndProxy(remoteBaseUrl: String) {
        // Record
        val wireMockManager =
            WireMockManager.Builder(applicationContext, NetworkConstants.LOCALHOST_PORT).build()
        wireMockManager.startProxyAndRecord(remoteBaseUrl)
    }


    fun playbackRecordedMappings() {
        // Playback
        val wireMockManager =
            WireMockManager.Builder(applicationContext, NetworkConstants.LOCALHOST_PORT).build()
        wireMockManager.startPlayBack()
    }
}
